jupyter_ai_personas/pr_review_persona/ci_tools.py

```diff
@@ -0,0 +1,126 @@
+import os
+import json
+from agno.tools import Toolkit
+from agno.utils.log import logger
+import re
+from github import Github
+import requests
+from agno.agent import Agent
+
+class CITools(Toolkit):
+    def __init__(self, **kwargs):
+        self.github_token = os.getenv("GITHUB_ACCESS_TOKEN")
+
+        if not self.github_token:
+            logger.warning("GITHUB_ACCESS_TOKEN environment variable is not set. GitHub operations will be limited.")
+        super().__init__(name="ci_tools", tools=[
+            self.fetch_ci_failure_data,
+            self.get_ci_logs
+        ],  **kwargs)
+
+    def fetch_ci_failure_data(self, agent: Agent,  repo_url: str, pr_number: int) -> list:
```

Comment:
It might be simpler to just expect a repo: `owner/repo` format than the URL, so that this code doesn't have to keep parsing the URL.